.github/workflows/ci.yml

```diff
@@ -0,0 +1,90 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: "43 3 * * *"
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    if: github.event_name == 'schedule' || github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Cache cargo
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+          key: cargo-cache-ubuntu-latest-${{ hashFiles('Cargo.toml') }}
```

Comment:
Are the cache results from the `lint`, `docs`, and `test` jobs identical? If not, we will store whichever was last to succeed (or first? - I'm not sure) which may not be what we want. Maybe we simply add the job name as a key component?

Reply:
Hmm they will have a lot of common files but won't be identical so yes we should have separate caches, I'll fix that.